### DISCUSSION TITLE
Project - Visualization API with Matplotlib

### DISCUSSION THREAD

**Discussion by HarshKhilawala** [networkx/networkx#6463] [General]
Hi 👋, I am new to NetworkX and would like to be a part of this community contributing to it. I am interested to work on one of the GSoC 2023 idea - **Project:- Visualization API with Matplotlib**. Before directly jumping onto the project proposal, I would like to get a bit of an understanding of the codebase and its working. I am a quick learner and have a great desire to grow as a software developer professional in the tech industry. Any hints on getting started with contributing to NetworkX would be appreciated.

@dschult Shall I start with the project proposal after setting up the environment locally? Any specific procedure to follow applying for GSoC Project?

**Comment by dschult** [comment]
Hi @HarshKhilawala !

To get acquainted with our development process please go through https://networkx.org/documentation/latest/developer/contribute.html#dev-workflow to setup
your local dev environment and https://networkx.org/documentation/latest/developer/new_contributor_faq.html for
some common FAQs.

For some issues to work on to get familiarised with NetworkX you can look at
the issues on the GitHub issue tracker https://github.com/networkx/networkx/issues. 
And some are "labeled" as "good first issue" or "sprint" or other names. But many haven't been labeled yet.

Please go through all the details available at https://summerofcode.withgoogle.com/ regarding application process.
You can look at a sample template at https://github.com/numfocus/gsoc/blob/master/templates/proposal.md
